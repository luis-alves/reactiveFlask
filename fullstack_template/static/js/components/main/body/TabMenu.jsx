import React from "react"

import { Tabs, Tab, TabList, TabPanel } from "react-tabs"
import Body from "./Body"
import '../../../../css/body/react-tabs.css';

class TabMenu extends React.Component {
  render () {
    return (
      <Tabs>
        <TabList>
          <Tab>Account</Tab>
          <Tab >Graphs</Tab>
        </TabList>

        <TabPanel>
          <Body />
        </TabPanel>
        <TabPanel>
          <p>test</p>
        </TabPanel>
      </Tabs>
    )
  }
}

export default TabMenu
