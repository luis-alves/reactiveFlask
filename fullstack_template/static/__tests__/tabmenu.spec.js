import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import TabMenu from '../js/components/main/body/TabMenu.jsx';

describe('TabMenu component', () => {
  it('renders without exploding', () => {
    expect(
      shallow(<TabMenu />).length).toEqual(1)
  })
  // it('Body has class bodymain', () => {
  //   const body = shallow(<Body />);
  //   expect(body.find('.bodymain').length).toEqual(1);
  // })
})

test('Check snapshot', () => {
  const wrapper = shallow(<TabMenu />);
  expect(wrapper).toMatchSnapshot();
});
