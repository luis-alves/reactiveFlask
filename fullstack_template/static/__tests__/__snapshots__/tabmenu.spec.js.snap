// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check snapshot 1`] = `
<Tabs
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="react-tabs__tab-list"
  >
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Account
    </Tab>
    <Tab
      className="react-tabs__tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      panelId={null}
      selected={false}
      selectedClassName="react-tabs__tab--selected"
    >
      Graphs
    </Tab>
  </TabList>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <Body />
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    selectedClassName="react-tabs__tab-panel--selected"
  >
    <p>
      test
    </p>
  </TabPanel>
</Tabs>
`;
